src/Crate/DBAL/Platforms/CratePlatform.php

```diff
@@ -656,9 +656,77 @@ protected function _getCreateTableSQL($tableName, array $columns, array $options
         }
 
         $query = 'CREATE TABLE ' . $tableName . ' (' . $columnListSql . ')';
+        $query .= $this->buildShardingOptions($options);
+        $query .= $this->buildPartitionOptions($options);
+        $query .= $this->buildTableOptions($options);
         return array($query);
     }
 
+    /**
+     * Build SQL for table options
+     *
+     * @param mixed[] $options
+     *
+     * @return string
+     */
+    private function buildTableOptions(array $options)
+    {
+        if (! isset($options['table_options'])) {
+            return '';
+        }
+
+        $tableOptions = [];
+        foreach($options['table_options'] as $key => $val) {
+            $tableOptions[] = sprintf('"%s" = %s', $key, $this->quoteStringLiteral($val));
+        }
+        if (count($tableOptions) == 0) {
+            return '';
+        }
+
+        return sprintf(' WITH (%s)', implode(', ', $tableOptions));
+    }
+
+    /**
+     * Build SQL for sharding options.
+     *
+     * @param mixed[] $options
+     *
+     * @return string
+     */
+    private function buildShardingOptions(array $options)
+    {
+        $shardingOptions = [];
+
+        if (isset($options['sharding_column'])) {
+            $shardingOptions[] = sprintf("BY (%s)", $options['sharding_column']);
+        }
+        if (isset($options['sharding_shards'])) {
+            $shardingOptions[] = sprintf("INTO %d SHARDS", $options['sharding_shards']);
+        }
+
+        if (count($shardingOptions) == 0) {
+            return '';
+        }
+
+        return sprintf(" CLUSTERED %s", implode(' ', $shardingOptions));
+    }
+
+    /**
+     * Build SQL for partition options.
+     *
+     * @param mixed[] $options
+     *
+     * @return string
+     */
+    private function buildPartitionOptions(array $options)
+    {
+        if (! isset($options['partition_columns'])) {
+            return '';
+        }
+
+        return sprintf(" PARTITIONED BY (%s)", $options['partition_columns']);
```

Comment:
I find passing a comma separated string a bit weird. I would have expected an array of columns as input.

Reply:
Also, I think the columns must be quoted identifiers if necessary.

```suggestion
        return sprintf(" PARTITIONED BY (%s)", (', ', $quoted_partition_columns));
```